todo-service/src/main/java/com/brtvsk/todoservice/model/entity/Todo.java

```diff
@@ -0,0 +1,112 @@
+package com.brtvsk.todoservice.model.entity;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.Date;
+import java.util.Set;
+import java.util.UUID;
+import java.util.Objects;
+
+
+@Document("todos")
+public class Todo {
+
+    @Id
+    private UUID id;
+    private String title;
+    private String description;
+    private Boolean done;
+    private Set<String> tags;
+    private Date creationTime;
+    private Date completionTime;
+
+    public Todo() {
+    }
+
+    public UUID getId() {
+        return id;
+    }
+
+    public void setId(final UUID id) {
+        this.id = id;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(final String title) {
+        this.title = title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(final String description) {
+        this.description = description;
+    }
+
+    public Boolean isDone() {
+        return done;
+    }
+
+    public void setDone(final Boolean done) {
+        this.done = done;
+    }
+
+    public Set<String> getTags() {
+        return tags;
```

Comment:
it better to return a copy of Set. F.e. `Set.of`

Reply:
Return Set.copyOf(tags)